src/renate/updaters/experimental/gdumb.py

```diff
@@ -56,9 +56,10 @@ def __init__(
 
     def load_state_dict(self, model: RenateModule, state_dict: Dict[str, Any], **kwargs) -> None:
         """Restores the state of the learner."""
+        if not hasattr(self, "_memory_buffer"):
+            self._memory_buffer = GreedyClassBalancingBuffer()
```

Comment:
is this happening because we create an instance without actually calling the constructor?

Reply:
Yes, as you can see in the old version, we already had to create a new buffer object in `load_state_dict` for precisely that reason. The `if` clause makes sure this only happens when we re-create the learner, not when we're just reloading an old state. (That's necessary because the new buffer implementation relies on persistence of its internal state when `load_state_dict` is called without `load`.)